src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -16,6 +18,78 @@ def filter_notification_link(link_text, amount, filter_item)
     end
   end
 
+  def request_badge_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined, :revoke
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    else
+      'dark'
+    end
+  end
+
+  def notification_link(notification)
+    link_text = case notification.event_type
+                when 'Event::RequestStatechange', 'Event::RequestCreate', 'Event::ReviewWanted'
+                  "Request ##{notification.notifiable.number}"
+                when 'Event::CommentForRequest'
+                  "Request ##{notification.notifiable.commentable.number}"
+                when 'Event::CommentForProject'
+                  notification.notifiable.commentable.name
+                when 'Event::CommentForPackage'
+                  commentable = notification.notifiable.commentable
+                  "#{commentable.project.name} / #{commentable.name}"
+                end
+
+    link_to(link_text, notification.link_to_notification_target, class: 'mx-1 text-word-break-all')
+  end
+
+  def read_unread_link(notification)
+    title, icon = notification.unread? ? ['Mark as "Read"', 'fa-check'] : ['Mark as "Unread"', 'fa-undo']
+
+    link_to(my_notification_path(id: notification),
```

Comment:
This can go to the view if `read_unread_link` just returns the array of `title` and `icon`.

Reply:
Everything has been moved to the view.